src/java.base/share/classes/sun/util/locale/provider/LocaleServiceProviderPool.java

```diff
@@ -138,15 +140,26 @@ private static class AllAvailableLocales {
                     LocaleServiceProviderPool.getPool(c);
                 all.addAll(pool.getAvailableLocaleSet());
             }
-
-            allAvailableLocales = all.toArray(new Locale[0]);
+            AvailableLocalesSet = all;
+            AvailableLocalesArray = all.toArray(new Locale[0]);
         }
 
         // No instantiation
         private AllAvailableLocales() {
         }
     }
 
+    /**
+     * Returns a stream of available locales for all the provider classes.
+     * This stream is constructed from all the locales
+     * that are provided by each provider, including the JRE.
+     *
+     * @return a stream of the available locales for all provider classes
+     */
+    public static Stream<Locale> streamAvailableLocales() {
+        return AllAvailableLocales.AvailableLocalesSet.stream();
```

Comment:
I think we could use `Stream.of(AllAvailableLocales.availableLocalesArray)` here and drop set field.

Reply:
Yes you're right, I had misinterpreted the original bug description. Set is not required, and in fact the array should be streamed directly as you said.